frontend/src/lib/utils.ts

```diff
@@ -20,3 +20,40 @@ export function splitSocketAddress(socketAddress: string) {
   const port = socketAddress.slice(lastColonIndex + 1);
   return { address, port };
 }
+
+export function generatePageNumbers(currentPage: number, totalPages: number) {
+  const MAX_PAGES_TO_SHOW = 3;
```

Comment:
Is this from shadcn? I didn't see it in the manual setup guide

Reply:
No, `shadcn` just gives the components for the pagination, the logic should be implemented by us (ig they did that for flexibility)